src/http_protocol.cpp

```diff
@@ -31,6 +33,27 @@ std::string url_decode(const std::string& encoded) {
   return decoded_str;
 }
 
+bool is_method_allowed(uint8_t mask, method_t method) {
+  return mask & static_cast<uint8_t>(method);
+}
+
+
+std::string get_allowed_methods_string(uint8_t verb_mask) {
+  std::string methods;
+  bool first = true;
+  for (int i = 0; i < 8; ++i) {
+    method_t method = static_cast<method_t>(pow(2, i));
+    if (is_method_allowed(verb_mask, method)) {
+      if (!first) {
+        methods += ", ";
+      }
+      methods += prime_server::METHOD_TO_STRING.find(method)->second;
+      first = false;
```

Comment:
you can do this without keeping `first` around because `methods` starts empty

```suggestion
      if (!methods.empty()) {
        methods += ", ";
      }
      methods += prime_server::METHOD_TO_STRING.find(method)->second;
```